### PR TITLE
feat(StyledComponents): Create `GlobalStyleProvider`

### LINKED ISSUE #2344: `styled-components` Context Provider (CSS Framework deprecation)
As we are moving the Design System to Styled Components, we should look to deprecate the **CSS Framework** package. In order to do so, we should create a `<ContextProvider>` component that wraps applications using the RNDS. This provider would contain everything in the CSS framework that isn't currently covered by Styled Components.

## Acceptance criteria
- [x] Investigated usage of utility classes in downstream applications 
- [x] `<ContextProvider>` component created, containing:
  - [x] `@font-face` rules to load `Lato`
  - [x] Resets
  - [x] Base styles (Normalize)
- [x] CSS Framework marked as Deprecated

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #2344

## Overview

Creates and exports a context provider that applies global styles to an RNDS application.

```jsx
import React from 'react'
import ReactDOM from 'react-dom'
import { GlobalStyleProvider, Button } from '@royalnavy/react-component-library'

function App() {
  return (
    <GlobalStyleProvider>
      <Button variant="primary">
        Hello, World!
      </Button>
    </GlobalStyleProvider>
  )
}

ReactDOM.render(<App />, document.querySelector('#app'))
```

## Reason

>As we are moving the Design System to Styled Components, we should look to deprecate the CSS Framework package. In order to do so, we should create a <ContextProvider> component that wraps applications using the RNDS. This provider would contain everything in the CSS framework that isn't currently covered by Styled Components.

## Work carried out 

- [x] Expose `<GlobalStyleProvider />`
- [x] Update CRA template
- [x] Test against dummy application (spawned from RN CRA template)
- [x] Add documentation to docs-site and README

## Developer notes

Considered writing a Cypress test that spawns a new application via the CRA template.

Holding off this for now, might be a way of expanding upon the existing smoke test.

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
It's worth noting that the generated helper classes currently remain in the CSS Framework package.

Not quite sure what to do with these just yet...

**Comment by m7kvqbe1** [comment]
Screenshot 2021-05-21 at 12 02 10

File: packages/react-component-library/src/styled-components/GlobalStyle.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import { selectors, BreakpointSize } from '@royalnavy/design-tokens'
+import '@royalnavy/fonts'
+
+interface GlobalStyleContextDefaults {
+  theme?: string | null
+}
+
+interface GlobalStyleProviderProps {
+  children?: React.ReactNode
+  theme?: string | null
```
Comment (thyhjwb6) [inline packages/react-component-library/src/styled-components/GlobalStyle.tsx]: is `null` needed when `theme` is nullable?

File: packages/react-component-library/src/styled-components/GlobalStyle.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import React, { createContext } from 'react'
+import { createGlobalStyle } from 'styled-components'
+import { Normalize } from 'styled-normalize'
+import { selectors, BreakpointSize } from '@royalnavy/design-tokens'
+import '@royalnavy/fonts'
+
+interface GlobalStyleContextDefaults {
+  theme?: string | null
```
Comment (thyhjwb6) [inline packages/react-component-library/src/styled-components/GlobalStyle.tsx]: See other comment about `null`

**Review by thyhjwb6 — APPROVED** [review]
Looks good, tested locally. 👍

**Pull request merged**